src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/SafeBufferTests.cs

```diff
@@ -76,6 +76,16 @@ public void ReadArray_NullArray_ThrowsArgumentNullException()
             AssertExtensions.Throws<ArgumentNullException>("array", () => wrapper.WriteArray<int>(0, null, 0, 0));
         }
 
+        [Fact]
+        public void ReadWriteSpan_EmptySpan_Passes()
+        {
+            var buffer = new SubBuffer(true);
+            buffer.Initialize(0);
+
+            buffer.ReadSpan<int>(0, Span<int>.Empty);
+            buffer.WriteSpan<int>(0, ReadOnlySpan<int>.Empty);
+        }
+
```

Comment:
We should have more tests to cover the new APIs than just this.

Reply:
Given that Read/WriteArray is delegated to the Span APIs, they automatically get covered by existing tests. I'm not sure duplicating the array tests is a good idea.

Reply:
For cases like this, we typically call the different entrypoints on the same test data. It minimizes duplication and keeps all public entry-points covered.

We do not seem to have any actual positive tests for the Read/Write methods. Could you please add some?

Reply:
Sure. Should be just round-tripping primitives and simple structs.

Reply:
It's not so clear to me how to create a valid `SafeBuffer`.
I can imagine `SetHandle(AllocHGlobal)`, `ReleaseHandle() => FreeHGlobal`, but how about the size? Nothing prevents `Initialize` from initializing more memory than allocated.